Nodejs/Product/Nodejs/Project/MigrateToJspsInfoBar.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudioTools;
+
+namespace Microsoft.NodejsTools.Project
+{
+    internal static class MigrateToJspsInfoBar
+    {
+        private static IVsInfoBarUIElement CurrentInfoBarElement;
+
+        public static void Show(IVsShell vsShell, IVsInfoBarUIFactory infoBarUiFactory)
+        {
+            if (ErrorHandler.Failed(vsShell.GetProperty((int)__VSSPROPID7.VSSPROPID_MainWindowInfoBarHost, out var tmp)))
+            {
+                return;
+            }
+            var infoBarHost = (IVsInfoBarHost)tmp;
+
+            CurrentInfoBarElement?.Close();
+
+            var infoBarModel = new InfoBarModel(Resources.MigrateToJspsPrompt);
+
+            uint eventCookie = 2;
+            CurrentInfoBarElement = infoBarUiFactory.CreateInfoBar(infoBarModel);
+            CurrentInfoBarElement.Advise(new InfoBarUIEvents(OnClose), out eventCookie);
+
+            infoBarHost.AddInfoBar(CurrentInfoBarElement);
+
+            void OnClose()
+            {
+                CurrentInfoBarElement.Unadvise(eventCookie); 
```

Comment:
If the user dismisses the gold bar we should probably not show it again the next time they open the project. We have some sample code in the JSPS repo (see SdkUpdatePrompter for how to use the user file).

Reply:
I wasn't able to use the CPS tools in JSPS to read the XML file -- noting this down to fix in the next iteration.

Reply:
Turns out we don't need it! The cookie event will prevent the goldbar from showing up again if the user dismisses it.